src/main/scala/io/iohk/ethereum/ledger/BlockQueue.scala

```diff
@@ -1,17 +1,16 @@
 package io.iohk.ethereum.ledger
 
 import akka.util.ByteString
-import io.iohk.ethereum.domain.{Block, Blockchain}
+import io.iohk.ethereum.domain.{Block, Blockchain, ChainWeight}
 import io.iohk.ethereum.ledger.BlockQueue.{Leaf, QueuedBlock}
 import io.iohk.ethereum.utils.Config.SyncConfig
 import io.iohk.ethereum.utils.Logger
 
 import scala.annotation.tailrec
-
 import scala.collection.JavaConverters._
 object BlockQueue {
-  case class QueuedBlock(block: Block, totalDifficulty: Option[BigInt])
-  case class Leaf(hash: ByteString, totalDifficulty: BigInt)
+  case class QueuedBlock(block: Block, weight: Option[ChainWeight])
```

Comment:
Because we are queueing blocks whose parents may not be known (they may come out of order) and in that case the chain weight (total difficulty) is also not known. This may change when blocks arrive, consider these competing chains:

```
A <- B <- C <- D
         ^- E <- F <- G
```

If blocks `F` and `G` arrive before `E`, then we enqueue them, but their chain weights will not be know until `E` is also enqueued.